### PR TITLE
Add report threshold

### LINKED ISSUE #107: Report threshold
Return the post only if the number of reports < threshold

### PR DESCRIPTION
**Pull request by amalaabraham**
Resolves #107 

1. Return the post only if the number of reports < threshold
     - retrieve_posts API (Retrieve all posts with user id present in the database)
     - retrieve_all_posts API (Retrieve all posts from the database)
     - retrieve_post API (Retrieve a post with a matching ID)
2. Added threshold to an env

### CODE REVIEW

**Review by aitikgupta — APPROVED** [review]
Thank you so much! This looks awesome 🎉

**Review by bodhisha — APPROVED** [review]
Looks Good🎉

File: backend/app/server/controllers/post.py [hunk 1]
```diff
@@ -80,27 +81,29 @@ async def retrieve_posts(user_id: ObjectId):
     user = await users_collection.find_one({"_id": ObjectId(user_id)})
     posts_by_user = posts_collection.find({"user_id": ObjectId(user_id)})
     async for post in posts_by_user:
-        likes_on_post = await get_all_likes_on_post(post["_id"])
-        comments_on_post = await get_all_comments_on_post(post["_id"])
-        posts.append(post_helper(post, user, likes_on_post, comments_on_post))
+        if (post["report_counter"] <=  config["THRESHOLD"]):
```
Comment (aitikgupta) [inline backend/app/server/controllers/post.py]:
```suggestion
        if (post["report_counter"] <= config["THRESHOLD"]):
```

File: backend/app/server/controllers/post.py [hunk 2]
```diff
@@ -80,27 +81,29 @@ async def retrieve_posts(user_id: ObjectId):
...
 
 # Retrieve all posts from the database
 async def retrieve_all_posts():
     posts = []
     async for post in posts_collection.find():
-        user = await users_collection.find_one({"_id": post["user_id"]})
-        likes_on_post = await get_all_likes_on_post(post["_id"])
-        comments_on_post = await get_all_comments_on_post(post["_id"])
-        posts.append(post_helper(post, user, likes_on_post, comments_on_post))
+        if (post["report_counter"] <=  config["THRESHOLD"]):
```
Comment (aitikgupta) [inline backend/app/server/controllers/post.py]:
```suggestion
        if (post["report_counter"] <= config["THRESHOLD"]):
```

File: backend/app/server/controllers/post.py [hunk 3]
```diff
@@ -80,27 +81,29 @@ async def retrieve_posts(user_id: ObjectId):
...
+            comments_on_post = await get_all_comments_on_post(post["_id"])
+            posts.append(post_helper(post, user, likes_on_post, comments_on_post))
     return posts
 
 
 # Retrieve a post with a matching ID
 async def retrieve_post(post_id: str) -> dict:
     post = await posts_collection.find_one({"_id": ObjectId(post_id)})
-    if post:
+    if post and (post["report_counter"] <=  config["THRESHOLD"]):
```
Comment (aitikgupta) [inline backend/app/server/controllers/post.py]:
```suggestion
    if post and (post["report_counter"] <= config["THRESHOLD"]):
```

**Review by aitikgupta — CHANGES_REQUESTED** [review]
Flake8 errors:

image

**Review by aitikgupta — APPROVED** [review]
Awesomee! 🎉

**Pull request merged**